lib/script.js

```diff
@@ -182,7 +182,12 @@ function wrapAsyncFunctions(asyncFunctions, sandbox, events, done, sandboxRoot)
             }
           };
         } else {
-          args.push(function() {
+          // CONVENTION:
+          // functions that use promises start with the name 'promisedXXXX', add a callback counter for them
+          if ((/^promised/).test(asyncFunctions[k].name)) {
+            events.emit('addCallback');
+          }
+          args.push(function () {
             if (sandboxRoot._error) return;
             events.emit('finishCallback');
```

Comment:
Yes, this was like this from before, I didn't want to touch it because I don't know what the purpose is, maybe there is some quirk where it is needed. It seems on purpose but it may very have been a bug:

In deployd previous to this PR, If you called the internal client **without** ( eg `dpd.collection.get(123, null)` ) a callback, the callback count would indeed be wrong, it would keep substracting 1 on each call. 

However, because of line 187-188 in this PR, this is now fixed because we increment callbackCount anyway for *promised* methods. So a function returning promises will always increment the callback count, and when the promise is resolved, it will call the supplied callback to decrement the callback count.